js/modules/Search/GenericView.js

```diff
@@ -100,12 +100,20 @@ window.GLPI.Search.GenericView = class GenericView {
             </div>
             `);
             const bs_modal = new bootstrap.Modal(modal.get(0), {show: false});
-            modal.on('show.bs.modal', () => {
-                const params = JSON.parse(modal.attr('data-params'));
-                params['url'] = window.location.pathname + window.location.search;
-                modal.find('.modal-body').load(CFG_GLPI.root_doc + '/ajax/savedsearch.php', params);
-            });
-            bs_modal.show();
+
+            if (window.location.search.includes('reset=reset') || window.location.search.trim() === '') {
```

Comment:
I am not sure that this condition will match all problematic cases.

A solution could be to rebuild the expected parameters based on the current criteria form values instead of relying on `window.location.search`. It is probably simple to do as this computation is probably already done on the JS side.

Reply:
If the user does not check the criteria (press the search button before save), `window.location.search` is empty.

I can't see how to rebuild the expected parameters based on the current values in the criteria form.

Reply:
Building the query params for the search is already handled in the `refreshResults` function of `Table.js`.

However, this isn't intuitive UX. When you create a bookmark in a browser for example, it is for the page you are on, not the page you are typing in the address bar but haven't navigated to yet. In Firefox, they even hide the bookmark button when you are typing in the address bar.

Simply disabling the button to create a saved search when there are unsaved changes, and having a helpful tooltip on it, should be sufficient.